pages/index.tsx

```diff
@@ -104,11 +102,17 @@ function Home({
   // retrieve query parameter with name cityFilter to set input value
   const { cityFilter } = router.query;
   // created state cityFilterQuery and initialized with query param cityFilter
-  const [cityFilterQuery, setCityFilterQuery] = useState<string>(cityFilter);
```

Comment:
This `useState<string>(cityFilter);` was giving a typescript error. 
Because `const { cityFilter } = router.query` can mean that cityFilter is 

1. A string
2. An array of strings `string[]`
3. Or undefined